src/embed/main.js

```diff
@@ -97,20 +104,24 @@ function onRenderLoad(event) {
     // Attach to pause and play events, to notify the API.
     event.videoElement.addEventListener('pause', onPause);
     event.videoElement.addEventListener('play', onPlay);
+    event.videoElement.addEventListener('timeupdate', onGetCurrentTime);
+    event.videoElement.addEventListener('ended', onEnded);
   }
   // Hide loading indicator.
   loadIndicator.hide();
 
   // Autopan only on desktop, for photos only, and only if autopan is enabled.
-  if (!Util.isMobile() && !worldRenderer.sceneInfo.video &&
-      !worldRenderer.sceneInfo.isAutopanOff) {
+  if (!Util.isMobile() && !worldRenderer.sceneInfo.video && !worldRenderer.sceneInfo.isAutopanOff) {
     worldRenderer.autopan();
   }
 
   // Notify the API that we are ready, but only do this once.
   if (!isReadySent) {
     Util.sendParentMessage({
-      type: 'ready'
+      type: 'ready',
+      data: {
+        duration: event.videoElement.duration
```

Comment:
Because I tried to make sure it's been set ahead of time. It's the behavior loadedmetadata will produce so the intention was to mimic it